debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java

```diff
@@ -146,22 +146,33 @@ public Lsn getMaxLsn() throws SQLException {
         }, "Maximum LSN query must return exactly one value"));
     }
 
-    public MaxLsnResult getMaxLsnResult(boolean skipLowActivityLsnsEnabled) throws SQLException {
-        if (skipLowActivityLsnsEnabled) {
-            return prepareQueryAndMap(GET_MAX_LSN_SKIP_LOW_ACTIVTY, statement -> {
-            }, singleResultMapper(rs -> {
-                final MaxLsnResult ret = new MaxLsnResult(Lsn.valueOf(rs.getBytes(1)), Lsn.valueOf(rs.getBytes(2)));
-                LOGGER.trace("Current maximum transactional LSN is {}", ret);
-                return ret;
-            }, "Maximum transactional LSN query must return exactly one value"));
-        }
-        Lsn maxLsn = prepareQueryAndMap(GET_MAX_LSN, statement -> {
+    /**
+     * @return the log sequence number of the most recent transaction
+     *         that isn't further than {@code maxOffset} from {@code lastLsn}.
+     */
+    public Lsn getNthTransactionLsn(Lsn lastLsn, int maxOffset) throws SQLException {
+        String replacement = lastLsn.isAvailable() ? " AND start_lsn >= ?" : "";
+        return prepareQueryAndMap(GET_NTH_TRANSACTION_LSN.replace(STATEMENTS_PLACEHOLDER, replacement), statement -> {
+            statement.setInt(1, maxOffset);
```

Comment:
Should it be `maxOffset + 1`? If I ask for 10 LSNs I mean 10 new ones (excluding last processed LSN). Additionally this way `maxOffset = 1` won't cause infinite loop.

Reply:
Alternatively just omit the equals sign from `AND start_lsn >= ?`

Reply:
> Should it be `maxOffset + 1`? If I ask for 10 LSNs I mean 10 new ones (excluding last processed LSN).

Yes, but it's only needed when there is last LSN.



> Alternatively just omit the equals sign from `AND start_lsn >= ?`

That won't work exactly the same because if there are no new transactions, `start_lsn > ?` will return `NULL` which the connector will interpret as "CDC not enabled". That's why `>=`.